### DISCUSSION TITLE
increase FPS

### DISCUSSION THREAD

**Discussion by hassanSayedd** [ultralytics/yolov5#2743] [Q&A]
How can i increase FPS to be capable with low specification like raspberry pi 4 ?!
i trained YOLO5s but it has low FPS 
and if want to deal with only the return value from the model and i do not want th show it how can i do that
thanks in advance

**Comment by glenn-jocher** [comment]
@hassanSayedd for CPU-only deployment destinations like Pi 4 the best option at the moment may be ONNX export with ONNX runtime (you **may** see a 3-4x speedup according to some sources), or also pruning and quantization aware training (QAT), which we don't fully support yet but may in the future.

**Comment by glenn-jocher** [comment]
@hassanSayedd of course, a second option altogether is to train a smaller architecture, ie a YOLOv5xs size, by reducing the compound scaling constants in the YOLOv5s model and then training again. You'd have to experiment with the best tradeoff between reducing the two scaling constants here:
https://github.com/ultralytics/yolov5/blob/c03d590320ea875a9ce5288c077a9ce5c7a1c160/models/yolov5s.yaml#L1-L5

**Reply by hassanSayedd** [reply]
@glenn-jocher i have been reduced value of depth_multiple and width_multiple and pretrained the model but it does not have a great effect on frame process time is there another thing i can do ??
and for ONNX i followed the steps and i get .onnx model but i don't know how to load it to make inference on my model sorry this is my first time to deal with ONNX model so i need your help !
thank you

**Reply by glenn-jocher** [reply]
@hassanSayedd for ONNX model inference you require ONNX Runtime: https://github.com/microsoft/onnxruntime

To increase speeds per image use a larger batch size, a smaller image, a smaller model, or a faster GPU

**Reply by SassyM** [reply]
> @hassanSayedd of course, a second option altogether is to train a smaller architecture, ie a YOLOv5xs size, by reducing the compound scaling constants in the YOLOv5s model and then training again. You'd have to experiment with the best tradeoff between reducing the two scaling constants here:
> 
> https://github.com/ultralytics/yolov5/blob/c03d590320ea875a9ce5288c077a9ce5c7a1c160/models/yolov5s.yaml#L1-L5

Do we have to train from scratch if we change the depth_multiple and width_multiple of any existing model? I read in another issue, that weights may be imported if the layers have identical names and shapes (https://github.com/ultralytics/yolov5/issues/1934#issuecomment-760519625). Could you clarify this?

**Comment by xrstokes** [comment]
I'm getting 15fps on a jetson nano with..
depth_multiple: 0.16  # model depth multiple 
width_multiple: 0.23  # layer channel multiple
